include/xtensor-io/xio_zlib.hpp

```diff
@@ -0,0 +1,347 @@
+/***************************************************************************
+* Copyright (c) Wolf Vollprecht, Sylvain Corlay and Johan Mabille          *
+* Copyright (c) QuantStack                                                 *
+*                                                                          *
+* Distributed under the terms of the BSD 3-Clause License.                 *
+*                                                                          *
+* The full license is in the file LICENSE, distributed with this software. *
+****************************************************************************/
+
+#ifndef XTENSOR_IO_ZLIB_HPP
+#define XTENSOR_IO_ZLIB_HPP
+
+#include <fstream>
+
+#include "zlib.h"
+
+#include "xtensor/xadapt.hpp"
+#include "xtensor-io.hpp"
+#include "xfile_array.hpp"
+#include "xio_stream_wrapper.hpp"
+
+#define ZLIB_CHUNK 0x4000
```

Comment:
Defining constant with macro should be avoided unless we want to give the opportunity to the user to override them. In that case, we should check for the existence of `ZLIB_CHUNK` before defining it.

Otherwise, this should be defined as a constexpr variable where it is used. If used in multiple functions, this should be wrapped into a utility function.